mailpile/plugins/vcard_face.py

```diff
@@ -0,0 +1,57 @@
+import time
+
+from mailpile.i18n import gettext as _
+from mailpile.mailutils import Email
+from mailpile.plugins import PluginManager
+from mailpile.search import MailIndex
+from mailpile.vcard import MailpileVCard, VCardImporter, VCardLine
+
+
+_plugins = PluginManager(builtin=__file__)
+
+
+class FaceImporter(VCardImporter):
+    """
+    Get avatar from Face header.
+    """
+    FORMAT_NAME = 'Face'
+    FORMAT_DESCRIPTION = _('Get avatar from Face header.')
+    SHORT_NAME = 'face'
+    CONFIG_RULES = {
+        'active': [_('Enable this importer'), bool, True],
+    }
+    VCARD_TS = 'x-face-ts'
+    VCARD_IMG = ''
+
+    def get_vcards(self):
+        mail_index = MailIndex(self.session.config)
+        mail_index.load(self.session)
+        results = []
+
+        for vcard in self.session.config.vcards.values():
+            # Get latest message from vcard email address.
+            from_messages = list(mail_index.search(
+                self.session, ['from:' + vcard.email]).as_set())
+            if not from_messages:
+                continue
+            mail_index.sort_results(self.session, from_messages, 'date')
+            latest_message = Email(mail_index, from_messages.pop()).get_msg()
+
+            # Add face vcard.
+            face = mail_index.hdr(latest_message, 'face')
+            if face:
+                results.append(self._make_vcard(vcard.email, VCardLine(
+                    name='photo',
+                    value='data:image/png;base64,%s' % face,
```

Comment:
Can we add some sort of sanity check to prevent the most obvious attacks through this vector? Consider what sort of mischief people could get up to by sending e-mails with malicious Face: headers...

Possible attacks:

1. very large images, bloating the vcard
2. malformed content of some sort